### DISCUSSION TITLE
On a Mac, how do I get podman working with the docker related vscode extensions?

### DISCUSSION THREAD

**Discussion by benatshippabo** [containers/podman#12745] [Q&A]
Hi, so I ran the following commands:

```sh
# based on this blog post https://www.redhat.com/sysadmin/podman-mac-machine-architecture
brew install podman
pip3 install podman-compose
podman machine init
podman machine start

# to get the ssh path
podman system connection ls
```

And it seems to work fine when testing in the cli with `podman` and `podman-compose`, but when I try to integrate podman with the vscode Docker extension and the Remote Containers extension, that is where I had trouble finding documentation.

For the record, I have the following configs set in attempt to get everything working:

```json
{
  "docker.dockerPath": "podman",
  "docker.host": "ssh://core@localhost:50261",
  "remote.containers.dockerPath": "podman",
  "remote.containers.dockerComposePath": "podman-compose"
}
```

And then I restarted the extension host. But the Docker extension just keeps ~~timing out~~ getting a socket hang up when trying to retrieve the containers, any idea where I went wrong?

---

cross posted here as well https://github.com/microsoft/vscode-docker/issues/3367#issue-1094635851

edit: [someone at Microsoft recommended adding the private key to `ssh-add`](https://github.com/microsoft/vscode-docker/issues/3367#issuecomment-1005993552) since the extension uses [`dockerode`](https://github.com/apocas/dockerode), now it's just hanging up on the socket, but I feel we are going somewhere with this

**Comment by mheon** [comment]
@baude @ashley-cui PTAL

**Reply by benatshippabo** [reply]
Just a quick update, after adding the ssh private key through `ssh-add ~/.ssh/podman-machine-default` and then manually specifying the dockerode options in vscode to:

```json
{
  "docker.dockerodeOptions": {
    "socketPath": "ssh://core@localhost:50685/run/user/1000/podman/podman.sock"
  }
}
```

~~I was able to connect to the vm instance,~~ although it seems that Podman does not support the `context` command? What would be the Podman equivalent of this command?

![image](https://user-images.githubusercontent.com/66273043/148424419-7392d2b9-52ef-4dfd-804f-3c94bb263211.png)

```
❯ podman context ls --format="{{json .}}"
Error: unknown flag: --format
❯ podman context ls
Error: unrecognized command `podman context`
Try 'podman --help' for more information.
❯ podman context
Error: unrecognized command `podman context`
Try 'podman --help' for more information.
```

---

edit: specifying both `docker.host` and `docker.dockerodeOptions.socketPath` [allows the extension to skip the call on the `context` command](https://github.com/microsoft/vscode-docker/issues/3367#issuecomment-1006826438) The error has progressed to ENOENT:

![image](https://user-images.githubusercontent.com/66273043/148435903-ba76e755-c1fa-4ac4-a302-a4c4519d93f1.png)

**Reply by mheon** [reply]
Does not look like we do. Please open an issue about this.

**Reply by benatshippabo** [reply]
Done! #12810 #12811
